testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -1337,5 +1573,98 @@ TableInfoAssert hasFilters(String... filterTexts)
             }
             return this;
         }
+
+        TableInfoAssert hasViewText(String viewText)
+        {
+            assertThat(actual.getViewText()).hasValueSatisfying(sql -> assertThat(sql).isEqualToIgnoringWhitespace(viewText));
+            return this;
+        }
+
+        @SafeVarargs
+        final TableInfoAssert hasTableReferencesSatisfying(Consumer<TableReferenceInfo>... tableReferenceAssertions)
+        {
+            assertThat(actual.getReferenceChain()).hasSize(tableReferenceAssertions.length);
+            for (int i = 0; i < tableReferenceAssertions.length; i++) {
+                tableReferenceAssertions[i].accept(actual.getReferenceChain().get(i));
+            }
+            return this;
+        }
+    }
+
+    static class TableReferenceInfoAssert
+            extends AbstractAssert<TableReferenceInfoAssert, TableReferenceInfo>
+    {
+        TableReferenceInfoAssert(TableReferenceInfo actual)
+        {
+            super(actual, TableReferenceInfoAssert.class);
+        }
+
+        BaseViewReferenceInfoAssert asViewInfo(boolean isMaterialized)
```

Comment:
?Can we introduce a new assertion like `asViewInfor` and as `materializedViewInfo` ?